service/src/main/java/uk/gov/hmcts/reform/fpl/service/MigrateCaseService.java

```diff
@@ -701,13 +702,19 @@ public Map<String, Object> removeCorrespondenceDocument(CaseData caseData,
                 .filter(el -> !expectedDocumentId.equals(el.getId()))
                 .collect(toList());
 
+        List<Element<SupportingEvidenceBundle>> newCorrespondenceDocumentsNC =
+            newCorrespondenceDocuments.stream()
+                .filter(el -> isNull(el.getValue().isConfidentialDocument()) || !el.getValue().isConfidentialDocument())
+                .collect(toList());
```

Comment:
Based on personal experience with a ticket few days ago, Sonar will raise this as code smell and fix for that is to use `.toList()` instead of `.collect(toList())` 🙂